41 new feature read csv datasets (#76)
umr-amap/StormR: vignettes/DataSource.Rmd

```

Before running `StormR` functions users have to provide a tropical storm track
dataset either as a ".nc" (NetCDF) or a ".csv" (CSV) file in which the location and some characteristics

> of storms are given across their lifespan. Using a CSV file as a dataset implies following few specific formatting rules, please refer to the "CSV formatting" section below. These files are then used to create a

Reviewer: Use RMrkdown linking feature (cf. https://bookdown.org/yihui/bookdown/cross-references.html)
```suggestion
of storms are given across their lifespan. Using a CSV file as a dataset implies following few specific formatting rules, please refer to the [CSV Formatting] section below. These files are then used to create a
```

`psi2pa`: converts psi in Pascal <br />
`None`: no conversion <br />

### Test data sets

> Two `test_dataset` are provided with the `StormR` package (a NetCDF and a CSV file). These test data sets

Reviewer: ```suggestion
A `test_dataset` is provided with the `StormR` package in two formats (a NetCDF and a CSV file). This test data set
```

comprises the track data of nine storms that occurred near Vanuatu and New
Caledonia between 2015-2016 and 2020-2021, respectively.

### CSV Formatting

In order to load and read a CSV file in `defStormsDataset()`, it must stick to the following rules:<br />
* As for the NetCDF file, the CSV file must contains at least a name, season, iso times, longitude, latitude,and a maximum wind speed column.
* Every observation (row) in the CSV must be sorted in increasing order according to the ISO times AND joined by storm. For example, if several storms happened during the same iso Times, the observations for the storm that first appeared should be stacked before the observations of the second one and so on.

In addition to these mandatory rules above, the default separator is set to a single comma ','. Nevertheless th user might decide to change it using the `sep` parameter in `defStormsDataset()`.

> 

Reviewer: Should include an example on how to access the sample dataset, and also a header of the file so that the user can see the link between the header of the file and the `fields` definition

Author: Done

`psi2pa`: converts psi in Pascal <br />
`None`: no conversion <br />

### Test data sets

A `test_dataset` is provided with the `StormR` package in two formats (a NetCDF and a CSV file). This test data set
comprises the track data of nine storms that occurred near Vanuatu and New
Caledonia between 2015-2016 and 2020-2021, respectively.

### CSV Formatting

In order to load and read a CSV file in `defStormsDataset()`, it must stick to the following rules:<br />
* As for the NetCDF file, the CSV file must contains at least a name, season, iso times, longitude, latitude,and a maximum wind speed column.
* Every observation (row) in the CSV must be sorted in increasing order according to the ISO times AND joined by storm. For example, if several storms happened during the same iso Times, the observations for the storm that first appeared should be stacked before the observations of the second one and so on.

In addition to these mandatory rules above, the default separator is set to a single comma ','. Nevertheless the user might decide to change it using the `sep` parameter in `defStormsDataset()`.

> Hereafter, an example where we access the data with `test_dataset.csv`.
> Notice that the header of the csv and the names of columns in `fields` input must be equals
> ```{r chunk 4}
> # Header of the csv
> head(read.csv(system.file("extdata", "test_dataset.csv", package = "StormR")))
> ```                         
> 
> 
> ```{r chunk 5}
> # Is already the default setting (in this particular case)
> fields <- c(
>   names = "name",
>   seasons = "season",
>   isoTime = "iso_time",
>   lon = "usa_lon",
>   lat = "usa_lat",
>   msw = "usa_wind",
>   sshs = "usa_sshs",
>   rmw = "usa_rmw",
>   pressure = "usa_pres",
>   poci = "usa_poci"
> )

Reviewer: ```suggestion
Hereafter, you can see the header of the dataset sample `test_dataset.csv` embbeded in the package.
Notice that the first line of the file must contain the names of columns and correspond to the `fields` input.
```{r chunk 4}
# Header of the csv
head(read.csv(system.file("extdata", "test_dataset.csv", package = "StormR")))
```                         
You can then define de `stormsDataset` object as follows:

```{r chunk 5}
# Is already the default setting (in this particular case)
fields <- c(
  names = "name",
  seasons = "season",
  isoTime = "iso_time",
  lon = "usa_lon",
  lat = "usa_lat",
  msw = "usa_wind",
  sshs = "usa_sshs",
  rmw = "usa_rmw",
  pressure = "usa_pres",
  poci = "usa_poci"
)
```

> SP_2015_2020_csv <- defStormsDataset(fields = fields, seasons = c(2015, 2020))

Reviewer: ```suggestion
SP_2015_2020_csv <- defStormsDataset(system.file("extdata", "test_dataset.csv", package = "StormR"), fields = fields, seasons = c(2015, 2020))
```
